peekingduck/pipeline/nodes/model/osnetv1/osnet_model.py

```diff
@@ -0,0 +1,216 @@
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+
+"""
+OSNet Inference.
+"""
+
+from typing import Any, Dict
+from pathlib import Path
+import os
+import logging
+import numpy as np
+from scipy.optimize import linear_sum_assignment
+from peekingduck.weights_utils import checker, downloader, finder
+from peekingduck.pipeline.nodes.model.osnetv1.osnet_files.crop import crop_bbox
+from peekingduck.pipeline.nodes.model.osnetv1.osnet_files.distance import (
+    compute_distance_matrix,
+)
+from peekingduck.pipeline.nodes.model.osnetv1.osnet_files.feature_extractor import (
+    FeatureExtractor,
+)
+
+
+class OSNetModel:  # pylint: disable=too-few-public-methods
+    """OSNet re-ID model for person identification."""
+
+    def __init__(self, config: Dict[str, Any]) -> None:
+        super().__init__()
+        self.logger = logging.getLogger(__name__)
+
+        # Check threshold values
+        if not 0 <= config["multi_threshold"] <= 1:
+            raise ValueError("multi_threshold must be in [0, 1]")
+
+        query_root_dir = config["query_root_dir"]
+        self.multi_threshold = config["multi_threshold"]
+        model_names = {
+            "osnet": "osnet_x1_0",
+            "osnet_ain": "osnet_ain_x1_0",
+        }
+
+        # Get names of folders in root query directory
+        persons = os.listdir(query_root_dir)
+        if not persons:
+            raise ValueError(
+                "There must be at least 1 folder in query_root_dir with \
+                images to be queried."
+            )
+
+        weights_dir, model_dir = finder.find_paths(
+            config["root"], config["weights"], config["weights_parent_dir"]
+        )
+
+        # check for OSNet weights, if none then download
+        if not checker.has_weights(weights_dir, model_dir):
+            self.logger.info("No weights detected. Proceeding to download...")
+            downloader.download_weights(weights_dir, config["weights"]["blob_file"])
+            self.logger.info(f"Weights downloaded to {weights_dir}.")
+
+        model_path = model_dir / config["weights"]["model_file"][config["model_type"]]
+        # Initialize model for extracting features
+        self.extractor = FeatureExtractor(
+            model_name=model_names[config["model_type"]],
+            model_path=str(model_path),
+            device=config["device"],
+        )
+        self.logger.info(
+            f"Model name: {model_names[config['model_type']]}\n"
+            " - params: 2,193,616\n"
+            " - flops: 978,878,352"
+        )
```

Comment:
Are there any other model config related values to be printed here? I see there's `multi_threshold` in the config file, should that be printed here with a proper description of what it stands for?

Reply:
Updated